src/components/EditImageModal/index.jsx

```diff
@@ -45,47 +43,48 @@ export default class EditImageModal extends React.Component {
 
     this.state = { ...initialEditImageModalState };
 
-    this.getStatusAlert = this.getStatusAlert.bind(this);
-    this.getStatusAlertDialog = this.getStatusAlertDialog.bind(this);
     this.handleOpenModal = this.handleOpenModal.bind(this);
     this.onConstrainProportionsClick = this.onConstrainProportionsClick.bind(this);
     this.onEditImageModalClose = this.onEditImageModalClose.bind(this);
     this.onImageDescriptionBlur = this.onImageDescriptionBlur.bind(this);
-    this.onImageIsDecorativeClick = this.onImageIsDecorativeClick.bind(this);
+    this.onImageDimensionChange = this.onImageDimensionChange.bind(this);
     this.onInsertImageButtonClick = this.onInsertImageButtonClick.bind(this);
+    this.onImageIsDecorativeClick = this.onImageIsDecorativeClick.bind(this);
     this.onImageLoad = this.onImageLoad.bind(this);
-    this.onImageSourceBlur = this.onImageSourceBlur.bind(this);
     this.onNextPageButtonClick = this.onNextPageButtonClick.bind(this);
     this.onPreviousPageButtonClick = this.onPreviousPageButtonClick.bind(this);
     this.onStatusAlertClose = this.onStatusAlertClose.bind(this);
-    this.validateImageDescription = this.validateImageDescription.bind(this);
-    this.validateImageSource = this.validateImageSource.bind(this);
     // Create ref setters to minimize anonymous inline functions
     this.setImageDescriptionInputRef = this.setImageDescriptionInputRef.bind(this);
     this.setImageFormRef = this.setImageFormRef.bind(this);
     this.setImagePreviewRef = this.setImagePreviewRef.bind(this);
-    this.setImageSourceInputRef = this.setImageSourceInputRef.bind(this);
     this.setModalWrapperRef = this.setModalWrapperRef.bind(this);
     this.setStatusAlertRef = this.setStatusAlertRef.bind(this);
     this.setPreviousButtonRef = this.setPreviousButtonRef.bind(this);
 
     this.imageDescriptionInputRef = null;
     this.imageFormRef = null;
-    this.imageSourceInputRef = null;
     this.imgRef = null;
-    this.inputDescriptionRef = null;
-    this.modalWrapperRef = null;
     this.previousPageButtonRef = null;
+    this.modalWrapperRef = null;
     this.statusAlertRef = null;
   }
 
   componentDidMount() {
     this.modalWrapperRef.addEventListener('openModal', this.handleOpenModal);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selectedAsset.portable_url) {
+      this.setState({
+        imageSource: nextProps.selectedAsset.portable_url,
+      });
+    }
+  }
+
   componentDidUpdate(prevProps, prevState) {
     if (this.state.pageNumber === 1 && prevState.pageNumber === 2) {
-      this.imageSourceInputRef.focus();
+      // TODO: add focus
```

Comment:
Is this supposed to focus on the drop zone here?

Reply:
Maybe, but I don't have the drop zone on my branch, which is why I'm leaving this here. Does that seem okay?

Reply:
Oh yeah :+1:, I just wanted to make sure.